### PR TITLE
feat(i18n): add complex translations component

### PR DESCRIPTION
**Pull request by wescopeland**
This PR adds a `<Trans />` component, loosely based on the [react-i18next implementation](https://react.i18next.com/latest/trans-component).

The `<Trans />` component is designed to account for advanced interpolation and nesting situations like what is mentioned in https://github.com/RetroAchievements/RAWeb/pull/2767#discussion_r1811619134.

This PR implements the component and changes 1 appropriate use case as proof that the implementation details work (along with the 100% code coverage):

**Before (3 Keys)**
```tsx
<p>
  {t('This is your')} <span className="italic">{t('personal')}</span>{' '}
  {t('web API key. Handle it with care.')}
</p>
```
```json
"This is your": "This is your",
"personal": "personal",
"web API key. Handle it with care.": "web API key. Handle it with care.",
```

**After (1 Key)**
```tsx
<p>
  <Trans i18nKey="This is your <0>personal</0> web API key. Handle it with care.">
    {'This is your'} <span className="italic">{'personal'}</span> {'web API key. Handle it with care.'}
  </Trans>
</p>
```
```json
"This is your <0>personal</0> web API key. Handle it with care.": "This is your <0>personal</0> web API key. Handle it with care.",
```

Implements https://github.com/EugeneMeles/laravel-react-i18n/issues/24.

### CODE REVIEW

**Comment by wescopeland** [comment]
cc @Krylan This is the component we were discussing earlier.

**Comment by Jamiras** [comment]
What is the significance of the number? You use `<0>` here, and `<1>` and `<5>` in the linked PR.

**Comment by Jamiras** [comment]
Is whitespace significant?
`your <0>personal` vs `{'This is your'} <span className="italic">{'personal'}`
(The space is before the `<0>`, but after `'}` following `your` in the value)

**Comment by wescopeland** [comment]
> What is the significance of the number? You use <0> here, and <1> and <5> in the linked PR.

This is an element index. Under the current implementation, in the key it must always start at 0, and it must always increase sequentially (to reiterate, in the key). I've updated the component's documentation to reflect this.

---

> Is whitespace significant?

It isn't.

---

> How does this work with SOV languages?

Let's take this example:
```tsx
<Trans i18nKey="This is your <0>personal</0> web API key. Handle it with care.">
  This is your <strong>personal</strong> web API key. Handle it with care.
</Trans>
```

In _en_US.json_:
```json
{
    "This is your <0>personal</0> web API key. Handle it with care.": "This is your <0>personal</0> web API key. Handle it with care."
}
```
This renders:
Screenshot 2024-10-23 at 5 33 05 PM

In _pt_BR.json_:
```json
{
    "This is your <0>personal</0> web API key. Handle it with care.": "<0>personal</0> web API key belonging to you must be handled with care."
}
```
This renders:
Screenshot 2024-10-23 at 5 34 14 PM

---

> What if order matters?

Let's take this example:
```tsx
<Trans i18nKey="Click <0>here</0> for our <1>FAQ</1>.">
  Click <a href="#">here</a> for our <strong>FAQ</strong>.
</Trans>
```

In _en_US.json_:
```json
{
    "Click <0>here</0> for our <1>FAQ</1>.": "Click <0>here</0> for our <1>FAQ</1>."
}
```
This renders:
Screenshot 2024-10-23 at 5 38 32 PM

In _pt_BR.json_:
```json
{
    "Click <0>here</0> for our <1>FAQ</1>.": "Our <1>FAQs</1> are <0>here</0>."
}
```
This renders:
Screenshot 2024-10-23 at 5 38 24 PM

**Comment by wescopeland** [comment]
Also important to note. The inner content is considered the safe fallback when the `i18nKey` cannot be found for the user's current locale. For example:
```tsx
<Trans i18nKey="Click <0>here</0> for our <1>FAQ</1>.">
  Click <a href="#">here</a> for our <strong>FAQ</strong>.
</Trans>
```

I am currently using the pt_BR locale, and this key is not in _pt_BR.json_. As a result, I get a safe fallback:
Screenshot 2024-10-23 at 5 43 01 PM

File: resources/js/features/settings/components/KeysSectionCard/ManageWebApiKey.tsx [hunk 1]
```diff
@@ -82,8 +83,10 @@ export const ManageWebApiKey: FC = () => {
 
           <div>
             <p>
-              {t('This is your')} <span className="italic">{t('personal')}</span>{' '}
-              {t('web API key. Handle it with care.')}
+              <Trans i18nKey="This is your <0>personal</0> web API key. Handle it with care.">
+                {'This is your'} <span className="italic">{'personal'}</span>{' '}
```
Comment (Jamiras) [inline resources/js/features/settings/components/KeysSectionCard/ManageWebApiKey.tsx:87]: Is the trailing space still needed here?

**Review by Jamiras — APPROVED** [review]

**Pull request merged**